tests/langchain/test_langchain_databricks_dependency_extraction.py

```diff
@@ -179,3 +180,90 @@ def test_parsing_dependency_from_databricks_chat(monkeypatch: pytest.MonkeyPatch
     _extract_databricks_dependencies_from_chat_model(chat_model, d, resources)
     assert d.get(_DATABRICKS_CHAT_ENDPOINT_NAME_KEY) == ["databricks-llama-2-70b-chat"]
     assert resources == [DatabricksServingEndpoint(endpoint_name="databricks-llama-2-70b-chat")]
+
+
+@contextmanager
+def remove_langchain_modules():
+    prefixes_to_remove = [
+        "langchain.",
+        "langchain_community.embeddings.DatabricksEmbeddings",
+        "langchain_community.vectorstores.DatabricksVectorSearch",
+        "langchain_community.llms.Databricks",
+        "langchain_community.chat_models.ChatDatabricks",
+    ]
+    langchain_modules = {
+        k: v
+        for k, v in sys.modules.copy().items()
+        if any(k.startswith(prefix) for prefix in prefixes_to_remove)
+    }
+
+    for k in langchain_modules:
+        del sys.modules[k]
+
+    try:
+        yield
+    finally:
+        for k, v in langchain_modules.items():
+            if k not in sys.modules:
+                sys.modules[k] = v
+            else:
+                pass
+
+
+@pytest.mark.skipif(
+    Version(langchain.__version__) < Version("0.0.311"), reason="feature not existing"
+)
+def test_parsing_dependency_correct_loads_langchain_modules(monkeypatch: pytest.MonkeyPatch):
```

Comment:
Shall we add a case to show the module removal is able to capture the incorrect case:
```python
def bad_function():
  import langchain
  langchain.llms.Databricks
with remove_langchain_modules():
  intercept[ModuleNotFoundError] {
    bad_function()
  }
```